pkg/documents/storage/test_bucket.go

```diff
@@ -24,3 +27,24 @@ var (
 		return fileBucket(ctx, log.NewNopLogger(), dir, signer)
 	}
 )
+
+// NewTestBucket returns a new BucketFunc, along with the underlying "temp" directory.
+// It's the caller's responsibility to remove the temp dir when it's no longer needed.
+func NewTestBucket() (string, func() (*blob.Bucket, error)) {
```

Comment:
In Go 1.15+ we can accept `t *testing.T` to call `t.Cleanup(...)`- that lets us hook into test cleanup without making each test know to `defer os.RemoveAll(..)`. 

```
$ go doc testing T.Cleanup 
package testing // import "testing"

func (c *T) Cleanup(f func())
    Cleanup registers a function to be called when the test and all its subtests
    complete. Cleanup functions will be called in last added, first called
    order.
```